### PR TITLE
Remove marshalling code.

### PR DESCRIPTION
**Pull request by danilvpetrov**
#### What change does this introduce?

This PR removes marshalling code from the repository. This marshalling code is intended to go to [`marshalkit`](https://github.com/dogmatiq/marhsalkit) package.

Also this change removes any references and usage of `marhalkit`, `configkit`, and `dogma` packages that were previously used in marshalling code and envelope validation tests.

#### What issues does this relate to?

Partially addresses https://github.com/dogmatiq/marshalkit/issues/40.

#### Why make this change?

This change is required to remove unrelated marshalling code from the envelope spec definitions.

#### Is there anything you are unsure about?

No.

### CODE REVIEW

**Comment by danilvpetrov** [comment]
Looks like the coverage dropped significantly for the protobuf methods after the marshalling functions were removed 😞

**Comment by jmalloc** [comment]
> Looks like the coverage dropped significantly for the protobuf methods after the marshalling functions were removed 😞

That's okay, it's all generated code anyway, nothing we need to worry about covering.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,14 @@ The format is based on [Keep a Changelog], and this project adheres to
 [Keep a Changelog]: https://keepachangelog.com/en/1.0.0/
 [Semantic Versioning]: https://semver.org/spec/v2.0.0.html
 
+## [Unreleased]
+
+### Removed
+
+- **[BC]** Remove marshalling function `MarshalMessage()`, ` UnmarshalMessage()`,
+  `MarshalIdentity()`, `UnmarshalIdentity()`, `MarshalTime()`, and
+  `UnmarshalTime()`.
```
Comment (jmalloc) [inline CHANGELOG.md:18]:
```suggestion
- **[BC]** Remove `MarshalMessage()` and ` UnmarshalMessage()`
- **[BC]** Remove `MarshalIdentity()` and `UnmarshalIdentity()`
- **[BC]** Remove `MarshalTime()` and `UnmarshalTime()`
```

Might be better to split these out into separate lines for related pairs of functions.

**Comment by danilvpetrov** [comment]
Thanks!

**Pull request merged**